[zlib] Add pkgconfig file

c72578 opened the pull request:
**Describe the pull request**

- What does your PR fix? Fixes #
Install the `zlib.pc` file, which is generated during the CMake build.
Fix the pkgconfig file under Windows, where the lib is called `zlib`
instead of `libz` and add the postfix 'd' in case of a Windows debug
build.

- Which triplets are supported/not supported? Have you updated the CI baseline?
All triplets are supported

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

c72578 committed:
[zlib] Add pkgconfig file

Install the zlib.pc file, which is generated during the CMake build.
Fix the pkgconfig file under Windows, where the lib is called zlib
instead of libz and add the postfix 'd' in case of a Windows debug
build.

ras0219 commented on ports/zlib/portfile.cmake:
```diff
@@ -31,6 +31,17 @@ vcpkg_configure_cmake(
+# Install the pkgconfig files
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/zlib.pc "-lz" "-lzlib")
+    vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/zlib.pc "-lz" "-lzlibd")
```
This should check `VCPKG_BUILD_TYPE` to handle release/debug-only builds. If it is unspecified, build both -- otherwise only build `release` or `debug`.

See https://github.com/microsoft/vcpkg/blob/a0e0c57f865937cdda81c21ad9327e606f7dc930/scripts/cmake/vcpkg_configure_cmake.cmake#L286-L293 as an example

c72578 replied:
@ras0219 Thanks for your feedback. I have added a commit now, which adds the check of VCPKG_BUILD_TYPE to handle release/debug.

c72578 committed:
Check VCPKG_BUILD_TYPE to handle release/debug

NancyLi1013 commented on ports/zlib/portfile.cmake:
```diff
@@ -31,6 +31,28 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
 
+# Install the pkgconfig file
+if(NOT DEFINED VCPKG_BUILD_TYPE)
+    if(VCPKG_TARGET_IS_WINDOWS)
+        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/zlib.pc "-lz" "-lzlib")
+        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/zlib.pc "-lz" "-lzlibd")
+    endif()
+    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/zlib.pc DESTINATION ${CURRENT_PACKAGES_DIR}/lib/pkgconfig)
+    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/zlib.pc DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
+elseif(VCPKG_BUILD_TYPE STREQUAL "release")
+    if(VCPKG_TARGET_IS_WINDOWS)
+        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/zlib.pc "-lz" "-lzlib")
+    endif()
+    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/zlib.pc DESTINATION ${CURRENT_PACKAGES_DIR}/lib/pkgconfig)
+elseif(VCPKG_BUILD_TYPE STREQUAL "debug")
+    if(VCPKG_TARGET_IS_WINDOWS)
+        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/zlib.pc "-lz" "-lzlibd")
+    endif()
+    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/zlib.pc DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
+endif()
```
```suggestion
if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
    if(VCPKG_TARGET_IS_WINDOWS)
        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/zlib.pc "-lz" "-lzlib")
    endif()
    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/zlib.pc DESTINATION ${CURRENT_PACKAGES_DIR}/lib/pkgconfig)
  endif()
if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
    if(VCPKG_TARGET_IS_WINDOWS)
        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/zlib.pc "-lz" "-lzlibd")
    endif()
    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/zlib.pc DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
endif()
```

NancyLi1013 commented on ports/zlib/portfile.cmake:
```diff
@@ -31,6 +31,28 @@ vcpkg_configure_cmake(
+vcpkg_fixup_pkgconfig()
+
 file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/zlib RENAME copyright)
 
```
`vcpkg_test_cmake(PACKAGE_NAME ZLIB MODULE)` doesn't work now. Please remove it or comment it out.

c72578 replied:
Thanks, updated

c72578 committed:
Simplify check of VCPKG_BUILD_TYPE to handle release/debug

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

c72578 committed:
Update PR according to review comments

- Remove include(vcpkg_common_functions)
- Comment out vcpkg_test_cmake(PACKAGE_NAME ZLIB MODULE)

c72578 commented on ports/zlib/portfile.cmake:
```diff
@@ -31,6 +31,28 @@ vcpkg_configure_cmake(
 
```
OK, removed

c72578 commented:
@NancyLi1013, is the PR OK now from your side?

NancyLi1013 commented on ports/zlib/portfile.cmake:
```diff
@@ -31,10 +29,26 @@ vcpkg_configure_cmake(
+
+vcpkg_fixup_pkgconfig()
+
 file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/zlib RENAME copyright)
```
```suggestion
file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

NancyLi1013 committed:
Update ports/zlib/portfile.cmake

NancyLi1013 commented:
@c72578 
LGTM now. Thanks for your PR.
Since there is a regression on osx for `itk `that is not related with this PR. But it sill needs to resolve.
Once all CI checks have passed, this PR can be ready to merge.

PhoebeHui commented:
The itk failures will be fixed by https://github.com/microsoft/vcpkg/pull/11208

PhoebeHui commented on ports/zlib/CONTROL:
```diff
@@ -1,4 +1,4 @@
 Source: zlib
-Version: 1.2.11-6
+Version: 1.2.11-7
```
```suggestion
Version: 1.2.11
Port-Version: 7
```

PhoebeHui committed:
Update ports/zlib/CONTROL
